apps/prairielearn/src/question-servers/freeform.ts

```diff
@@ -1168,8 +1168,10 @@ export async function render(
         clientFilesCourseScripts: {} as Record<string, string>,
       };
 
-      for (const type in question.dependencies) {
-        if (!(type in dependencies)) continue;
+      for (const rawType in question.dependencies) {
```

Comment:
Typescript is really struggling with this code, and am doing lots of (hopefully safe) casts. Potentially a sign to rewrite in a style that it can handle better?